x-pack/plugins/fleet/dev_docs/developing_kibana_and_fleet_server.md

```diff
@@ -0,0 +1,278 @@
+
+This guide seeks to get you up and running with the following stack components for local development:
+
+- Kibana (from source)
+- Elasticsearch (from a snapshot)
+- Fleet Server (from source, in standalone mode)
+
+Getting this development environment up and running will allow you to make changes to both Kibana and Fleet Server simultaneously to develop and test features end-to-end.
+
+Before continuing, please review the developer documentation for both Fleet Server and Kibana. Make sure your local workstation can run each service independently. This means setting up Go, Node.js, Yarn, resolving dependencies, etc.
+
+- https://github.com/elastic/fleet-server?tab=readme-ov-file#development
+- https://github.com/elastic/kibana/blob/main/CONTRIBUTING.md
+
+## Start up Kibana and Elasticsearch
+
+1. Get a local Kibana checkout up and running by following [the developer guide](https://docs.elastic.dev/kibana-dev-docs/getting-started/setup-dev-env) (internal Elastic employee link for now). Run the commands below from your `kibana` directory
+2. Run a local Elasticsearch server from the latest snapshot build by running:
+
+```bash
+# Enable SSL + set a data path to prevent blowing away your Elasticsearch data between server restarts
+$ yarn es snapshot --license trial --ssl -E path.data=/tmp/es-data
+```
+
+1. Add the following to your `kibana.dev.yml`
+
+```bash
+server.basePath: '/some-base-path' # Optional
+server.versioned.versionResolution: oldest
+elasticsearch.hosts: [<http://localhost:9200>]
+
+# Optional - set up an APM service in a cloud.elastic.co cluster to send your dev logs, traces, etc
+# Can be helpful for troubleshooting and diagnosting performance issues
+# elastic.apm:
+  # active: true
+  # serverUrl: <https://some-cluster.apm.us-east4.gcp.elastic-cloud.com:443>
+  # secretToken: some-token
+  # breakdownMetrics: true
+  # transactionSampleRate: 0.1
+
+logging:
+  loggers:
+    - name: plugins.fleet
+      appenders: [console]
+      level: debug
+
+# Allows enrolling agents when standalone Fleet Server is in use
+xpack.fleet.internal.fleetServerStandalone: true
+
+xpack.fleet.fleetServerHosts:
+  - id: localhost
+    name: Localhost
+    # Make sure this is `https` since we're running our local Fleet Server with SSL enabled
+    host_urls: ['<https://localhost:8220>']
+ # If you want to run a Fleet Server in Docker, use this Fleet Server host
+  - id: docker
+    name: Docker Internal Gateway
+    host_urls: ['<https://host.docker.internal:8220>']
+    is_default: true
+
+xpack.fleet.packages:
+  - name: fleet_server
+    version: latest
+
+xpack.fleet.outputs:
+  - id: preconfigured-localhost-output
+    name: Localhost Output
+    type: elasticsearch
+    hosts: ['<https://localhost:9200>']
+    ca_trusted_fingerprint: 'f71f73085975fd977339a1909ebfe2df40db255e0d5bb56fc37246bf383ffc84'
```

Comment:
is this a static fingerprint or does it have to be generated?

Reply:
It's static based on the source of the SSL cert that's shipped with Kibana. If that cert changes it'd need to be recalculated.